applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c

```diff
@@ -110,62 +110,73 @@ static void accelerometer_trigger_handler(const struct device *dev,
 
 int ext_sensors_init(ext_sensor_handler_t handler)
 {
+	struct ext_sensor_evt evt = {0};
+
 	if (handler == NULL) {
 		LOG_INF("External sensor handler NULL!");
 		return -EINVAL;
 	}
 
+	evt_handler = handler;
+
 	if (!device_is_ready(temp_sensor.dev)) {
 		LOG_ERR("Temperature sensor device is not ready");
-		return -ENODEV;
+		evt.type = EXT_SENSOR_EVT_TEMPERATURE_DEVICE_ERROR;
+		evt_handler(&evt);
```

Comment:
**question:** Is this common practice to call a handler *multiple* times on error (in this case it can happen for each failed sensor)?

Reply:
If the errors are unrelated this can be done. For instance, in this case, we need to propagate separate error depending on the failing sensor.

Reply:
Sure, but it could also be done in one call (pseudo code follows):

```
evt.type = 0
...
// Create list of failed sensors
if (sensor1failure) evt.type = evt.type & ERROR_1_EVT;
if (sensor2failure) evt.type = evt.type & ERROR_2_EVT;
if (sensor3failure) evt.type = evt.type & ERROR_3_EVT;
...
// Call handler exactly once
if (evt.type != 0) evt_handler(&evt);
```

Reply:
I though of something like this initially and its certainly possible propagate a list of failed sensors. But I'm not completely sold on if it is worth the extra work. It looks like it would be only suited to be used in `ext_sensors_init` and it is most likely that the `sensor_module` is going to be reworked with the new CAF framework. (Which could mean that we wouldn't need the `ext_sensor` library at all.